tensorflow_addons/activations/hardshrink.py

```diff
@@ -14,9 +14,33 @@
 # ==============================================================================
 
 import tensorflow as tf
-from tensorflow_addons.utils.resource_loader import LazySO
 
-_activation_so = LazySO("custom_ops/activations/_activation_ops.so")
+
+def _hardshrink(x, lower, upper):
+    mask_lower = x < lower
+    mask_upper = upper < x
+    mask = tf.logical_or(mask_lower, mask_upper)
+    mask = tf.cast(mask, x.dtype)
+    return x * mask
+
+
+def compile_with_xla(func, dtype):
```

Comment:
I'm wondering if we could create a xla decorator here?

Reply:
I changed the code structure. Please have a look.